packages/core/src/payment/strategies/amazon-pay-v2/amazon-pay-v2-payment-processor.ts

```diff
@@ -109,30 +128,26 @@ export default class AmazonPayV2PaymentProcessor {
         if (features['PROJECT-3483.amazon_pay_ph4']) {
             const amount = getCheckout()?.outstandingBalance.toString();
             const currencyCode = cart?.currency.code;
-            const buttonOptions = /^(SANDBOX|LIVE)/.test(publicKeyId)
-                ? {
-                      ...buttonBaseConfig,
-                      publicKeyId,
-                      createCheckoutSessionConfig,
-                  }
-                : {
-                      ...buttonBaseConfig,
-                      sandbox: !!testMode,
-                      createCheckoutSessionConfig: {
-                          ...createCheckoutSessionConfig,
-                          publicKeyId,
-                      },
-                  };
-
-            return amount && currencyCode
-                ? {
-                      ...buttonOptions,
-                      estimatedOrderAmount: {
-                          amount,
-                          currencyCode,
-                      },
-                  }
-                : buttonOptions;
+            const buttonOptions: AmazonPayV2NewButtonParams = { ...buttonBaseConfig };
+            let tmpConfig: AmazonPayV2CheckoutSessionConfig;
+
+            if (this._isEnvironmentSpecific(publicKeyId)) {
+                buttonOptions.publicKeyId = publicKeyId;
```

Comment:
By reading this seems like based on the check `publicId` can be a part of `buttonOptions` or `createCheckoutSessionConfig` ?

Reply:
That's right.

<img width="1633" alt="v2 Flavors" src="https://user-images.githubusercontent.com/4843328/180506088-43f30791-063c-4856-a1f6-f3336671f2fd.png">